openlcs/libs/corgi.py

```diff
@@ -464,8 +464,16 @@ def get_container_source_components(self, component, subscribed_purls=None,
             link = sources[0].get("link")
             component = self.get(link, includes=self.oci_includes)
             logger.debug("Binary build %s retrieved.", component['nevra'])
-
-        oci_provides = component.get("provides", [])
+        if component.get('arch') == 'noarch':
+            oci_provides = component.get("provides", [])
+        else:
+            # If the binary container arch is not noarch, get the provides
+            # of the provides of the noarch container.
+            oci_provides = component.get("provides", [])
+            if component.get('sources'):
+                link = component.get('sources')[0].get("link")
+                noarch_oci = self.get(link, includes=self.oci_includes)
+                oci_provides = noarch_oci.get("provides", [])
```

Comment:
The noarch container's 'provides' includes those of its arch specific binary containers in Corgi, e.g., 

> 
    ubi8-container-8.7-1037.x86_64, 153 provides
    ubi8-container-8.7-1037.aarch64, 151 provides
    ubi8-container-8.7-1037.ppc64le, 151 provides
    ubi8-container-8.7-1037.s390x, 150 provides
    ubi8-container-8.7-1037.noarch, 640 provides(including 4 oci provides), parent of above 4 architecture-specific containers.
Using the 'provides' of noarch directly, means the corgi query count will be n-1(n means the arch types) times larger than expected. It would be better if this could be refined, no missing and no duplicate provide(same nvr but with different arches) which is same with that in source image.

Reply:
Updated. Please review it again. Thanks.